src/netif.c

```diff
@@ -1770,87 +1848,128 @@ static int get_default_gateway(const struct ip_setting_handler *handler, struct
-        const struct ip_setting_handler *handler = &handlers[i];
-        if (strcmp(handler->name, name) == 0)
-            return handler;
+  for (size_t i = 0; i < HANDLER_COUNT; i++) {
+    const struct ip_setting_handler *handler = &handlers[i];
+    if (strcmp(handler->name, name) == 0)
+      return handler;
   }
   return NULL;
 }
 
 static void netif_handle_set(struct netif *nb,
                              const char *ifname)
 {
-    nb->last_error = 0;
+  void *handler_context[HANDLER_COUNT] = { NULL, };
+  int arity = 0;
 
-    start_response(nb);
+  start_response(nb);
 
-    int arity;
-    if (ei_decode_map_header(nb->req, &nb->req_index, &arity) < 0)
-        errx(EXIT_FAILURE, "setting attributes requires a map");
-
-    void *handler_context[HANDLER_COUNT];
-    memset(handler_context, 0, sizeof(handler_context));
-
-    // Parse all options
-    for (int i = 0; i < arity && nb->last_error == 0; i++) {
-        char name[32];
-        if (erlcmd_decode_atom(nb->req, &nb->req_index, name, sizeof(name)) < 0)
-            errx(EXIT_FAILURE, "error in map encoding");
-
-        // Look up the option. If we don't know it, silently ignore it so that
-        // the caller can pass in maps that contain options for other code.
-        const struct ip_setting_handler *handler = find_handler(name);
-        if (handler)
-            handler->prep(handler, nb, &handler_context[handler - handlers]);
-        else
-            ei_skip_term(nb->req, &nb->req_index);
-        }
+  nb->last_error = 0;
 
-    // If no errors, then set everything
-    if (!nb->last_error) {
-        // Order is important: see note on handlers
-        for (size_t i = 0; i < HANDLER_COUNT; i++) {
-            if (handler_context[i]) {
-                handlers[i].set(&handlers[i], nb, ifname, handler_context[i]);
-                free(handler_context[i]);
-            }
-        }
+  if (ei_decode_map_header(nb->req, &nb->req_index, &arity) < 0)
+     errx(EXIT_FAILURE, "setting attributes requires a map");
+
+  // Parse all options
+  for (int i = 0; i < arity && nb->last_error == 0; i++) {
+    struct ip_setting_handler *handler = NULL;
+    char name[32] = {'\0', };
+
+    if (erlcmd_decode_atom(nb->req, &nb->req_index, name, sizeof(name)) < 0)
+      errx(EXIT_FAILURE, "error in map encoding");
+
+    // Look up the option. If we don't know it, silently ignore it so that
+    // the caller can pass in maps that contain options for other code.
+    handler = (struct ip_setting_handler *) find_handler(name);
+
+    debug("%s %d]: handler for '%s' = %p\r\n", __FILE__, __LINE__, name, (void *) handler);
+
+    if (handler != NULL) {
+      handler->prep(handler, nb, &handler_context[handler - handlers]);
+    } else {
+      ei_skip_term(nb->req, &nb->req_index);
+    }
+  }
+
+  // If no errors, then set everything
+  if (!nb->last_error) {
+    size_t i = 0;
+    // Order is important: see note on handlers
+    for (; i < HANDLER_COUNT; i++) {
```

Comment:
Same here.  Lets keep the loop variable in the `for()` 

```
for (size_t i = 0;  i < HANDLER_COUNT; i++)
```

Reply:
I'm with Mike on this one, it makes it more readable.

Reply:
this is actually the c++ thing: for (size_t i = 0;  i < HANDLER_COUNT; i++)
whereas ansi c would have rather
size_t i; /* left uninitialised */
for(i = 0; some cond; i++) { ... }